### DISCUSSION TITLE
Building a wheel with VCPKG dependencies on Linux CI

### DISCUSSION THREAD

**Discussion by integraledelebesgue** [pypa/cibuildwheel#2031] [Q&A]
Hi! 
I struggle with using vcpkg together with cibuildwheel on Linux CI job.
[My project](https://github.com/child-lab-uj/mini-face-meson-test) has a few elements:
1. A Python package, with C++ extension module which has some external C++ dependencies (dlib, OpenCV, OpenBLAS to name a few). Locally I manage them using vcpkg.
2. Meson build system - both as a Python backend and C++ build system. I have a hierarchical meson.build structure same as in e.g. NumPy.
3.  meson.build responsible for C++ extension module call vcpkg to install dependencies before compilation.
4. A CI in which I would like to build and publish wheels of my library. It installs vcpkg and cibuildwheel in particular

I try to build wheels using cibuildwheel in my CI. It works perfectly fine on macOS job, but fails on Linux because Meson is unable to locate vcpkg.
Somehow vcpkg isn't available in a cibuildwheel container in which build happens. I tried passing environmental variables using `environment`, `environment-pass` etc, even cloning and bootstrapping vcpkg as a `before-all` step, but nothing works.

May I ask for an advice how I should set up my build properly?

**Comment by henryiii** [comment]
It's in a container, so I'd expect the cloning-and-boostrapping one to work (assuming vcpkg on linux builds from source, rather than getting binaries, which probably would not be manylinux compatible).
